src/diart/websockets.py

```diff
@@ -0,0 +1,370 @@
+
+    This handler manages WebSocket connections from multiple clients, processing
+    audio streams and performing speaker diarization in real-time.
+
+    Parameters
+    ----------
+    pipeline_class : type
+        Pipeline class
+    pipeline_config : blocks.PipelineConfig
+        Pipeline configuration
+    host : str, optional
+        WebSocket server host, by default "127.0.0.1"
+    port : int, optional
+        WebSocket server port, by default 7007
+    key : Union[str, Path], optional
+        SSL key file path for secure WebSocket
+    certificate : Union[str, Path], optional
+        SSL certificate file path for secure WebSocket
+    """
+
+    def __init__(
+        self,
+        pipeline_class: type,
+        pipeline_config: blocks.PipelineConfig,
+        host: Text = "127.0.0.1",
+        port: int = 7007,
+        key: Optional[Union[Text, Path]] = None,
+        certificate: Optional[Union[Text, Path]] = None,
+    ):
+        # Pipeline configuration
+        self.pipeline_class = pipeline_class
+        self.pipeline_config = pipeline_config
+
+        # Server configuration
+        self.host = host
+        self.port = port
+        self.uri = f"{host}:{port}"
+        self._clients: Dict[Text, ClientState] = {}
+
+        # Initialize WebSocket server
+        self.server = WebsocketServer(host, port, key=key, cert=certificate)
+        self._setup_server_handlers()
+
+    def _setup_server_handlers(self) -> None:
+        """Configure WebSocket server event handlers."""
+        self.server.set_fn_new_client(self._on_connect)
+        self.server.set_fn_client_left(self._on_disconnect)
+        self.server.set_fn_message_received(self._on_message_received)
+
+    def _create_client_state(self, client_id: Text) -> ClientState:
+        """Create and initialize state for a new client.
+
+        Parameters
+        ----------
+        client_id : Text
+            Unique client identifier
+
+        Returns
+        -------
+        ClientState
+            Initialized client state object
+        """
+        # Create a new pipeline instance with the same config
+        # This ensures each client has its own state while sharing model weights
+        pipeline = self.pipeline_class(self.pipeline_config)
+
+        audio_source = ProxyAudioSource(
+            uri=f"{self.uri}:{client_id}", sample_rate=self.pipeline_config.sample_rate,
+        )
+
+        inference = StreamingInference(
+            pipeline=pipeline,
+            source=audio_source,
+            # The following variables are fixed for a client
+            batch_size=1,
+            do_profile=False,  # for minimal latency
+            do_plot=False,
+            show_progress=False,
+            progress_bar=None,
+        )
+
+        return ClientState(audio_source=audio_source, inference=inference)
+
+    def _on_connect(self, client: Dict[Text, Any], server: WebsocketServer) -> None:
```

Comment:
Maybe we should allow a max number of clients to connect? My reasoning is the following: if we have to copy `StreamingInference` instances (including models) for every new client, the server will most likely crash at some point (especially if sharing GPU). However, given system resources, we can probably estimate how many clients fit in the machine, or if the new client fits in the remaining available resources.

If this is too complicated, we can simply add a parameter inside `__init__()` for the maximum number of simultaneous clients. Something like `client_pool_size: int = 4`.